mod/primitives/pkg/encoding/hex/u64.go

```diff
@@ -37,16 +37,6 @@ func MarshalText(b uint64) ([]byte, error) {
 	return buf, nil
 }
 
-// ValidateUnmarshalInput validates the input byte slice for unmarshaling.
-// It returns an error iff input is not a quoted string.
-// This is used to prevent exposing validation logic to the caller.
-func ValidateUnmarshalInput(input []byte) error {
-	if !isQuotedString(string(input)) {
-		return ErrNonQuotedString
-	}
-	return nil
-}
-
```

Comment:
q: in `UnmarshalUint64Text`, I wonder if we should use `return strconv.ParseUint(string(raw), hexBase, 0)` instead of the current implementation?